src/umpire/resource/MemoryResourceTypes.hpp

```diff
@@ -18,7 +23,46 @@ struct MemoryResourceTypeHash {
   }
 };
 
-enum MemoryResourceType { Host, Device, Unified, Pinned, Constant };
+enum MemoryResourceType { Host, Device, Unified, Pinned, Constant, File };
+
+inline std::string resource_to_string(MemoryResourceType type)
+{
+  switch (type) {
+    case Host:
+      return "HOST";
+    case Device:
+      return "DEVICE";
+    case Unified:
+      return "UM";
+    case Pinned:
+      return "PINNED";
+    case Constant:
+      return "DEVICE_CONST";
+    case File:
+      return "FILE";
+    default: 
+      UMPIRE_ERROR("Unkown resource type: " << type);
+  }
+}
+
+inline MemoryResourceType string_to_resource(const std::string& resource)
+{
+  if (resource == "HOST") return MemoryResourceType::Host;
+  else if (resource == "DEVICE") return MemoryResourceType::Device;
+  else if (resource == "UM") return MemoryResourceType::Unified;
+  else if (resource == "PINNED") return MemoryResourceType::Pinned;
+  else if (resource == "DEVICE_CONST") return MemoryResourceType::Constant;
+  else if (resource == "FILE") return MemoryResourceType::File;
+  else UMPIRE_ERROR("Unkown resource name: " << resource);
+}
+
+inline int resource_to_device_id(const std::string& resource) {
+  int device_id{0};
+  if (resource.find("::") != std::string::npos) {
```

Comment:
Could use # as the separator.

Reply:
@davidbeckingsale, what is left to do on this?  Is it just replay?  If so, I can work on getting replay to work.